### PR TITLE
New package: dosbox-staging-0.77.0

### PR DESCRIPTION
**Pull request by joshuakraemer**
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me

This is a draft based on a current git snapshot, because DOSBox Staging 0.77.0 has not been released yet.

@kcgen wrote (https://github.com/joshuakraemer/void-packages/commit/2d0e6a7ee8328c8210c8ed272dc38dd8317296ff#commitcomment-51084600):
> @joshuakraemer, looks good!
> 
> I realize this is using the `master` branch right now until we can wrap up 0.77. In the meantime, regarding: `-Duse_mt32emu=false`, can you try removing that? Meson should be able to build it internally based on its wrap. There should be no copyright concerns, as users must provide their own ROMs for this feature.

Thank you for your comments. I have added libmt32emu as a separate package, and successfully tested MT-32 emulation.

> Also, there might be more configuration options needed to ensure an optimized binary is built. Here's what I use:
> 
> ```shell
> -Dbuildtype=release \
> -Dc_args=-Ofast \
> -Dcpp_args=-Ofast \
> -Db_asneeded=true -Dstrip=true \
> -Ddefault_library=static \
> -Dfluidsynth:enable-floats=true \
> -Dfluidsynth:try-static-deps=true 
> ```
> 
> Perhaps the static settings could be dropped depending on what void Linux prefers.
> I personally want to ensure that FluidSynth is compiled with the same optimization level (and uses our highly stripped down FluidSynth configuration) to ensure good performance on some of my CPU-limited platforms.

Void's build options for meson packages are specified in https://github.com/void-linux/void-packages/blob/master/common/build-style/meson.sh (using -O2 from the CFLAGS environment variable). The enable-floats option could probably be added to Void's FluidSynth package (https://github.com/void-linux/void-packages/blob/master/srcpkgs/fluidsynth/template), but I do not know if it would be a suitable default outside of DosBox.

### CODE REVIEW

**Comment by joshuakraemer** [comment]
Build fails with musl because the wrong strerror_r() variant is used. Upstream bug report: https://github.com/dosbox-staging/dosbox-staging/issues/1047

**Comment by kcgen** [comment]
> The enable-floats option could probably be added to Void's FluidSynth package (https://github.com/void-linux/void-packages/blob/master/srcpkgs/fluidsynth/template), but I do not know if it would be a suitable default outside of DosBox.

I highly recommend it across the board.  Floats are already 32-bits - which gives you 2^16 times finer resolution to quantify an amplitude level versus CD-quality 16-bit sound.

Even 24-bit audio is considered sufficient for professional-grade studio work. Leaving this as a double means samples are a whopping 64-bit; which is.. vastly overkill.   

The other benefit of 32-bit floats (versus full-doubles) is that SIMD'able operations can pack twice the data into a single operation, which can make  difference on CPU limited hardware like the Pi 2 and 3.  I make these two points in another thread and FluidSynth's author was also in agreement -  but can't remember where now.

File: srcpkgs/dosbox-staging/template [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Template file for 'dosbox-staging'
+pkgname=dosbox-staging
+version=0.77.0
+revision=1
+build_style=meson
+hostmakedepends="pkg-config ccache"
+makedepends="alsa-lib-devel libpng-devel SDL2-devel SDL2_net-devel opusfile-devel fluidsynth-devel libmt32emu-devel libatomic-devel"
```
Comment (ericonr) [inline srcpkgs/dosbox-staging/template:7]: Please create a new line at ~80 columns.

Is `libatomic` always necessary or only for `XBPS_TARGET_NO_ATOMIC8` archs (like armv6l)?

File: srcpkgs/dosbox-staging/template [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Template file for 'dosbox-staging'
+pkgname=dosbox-staging
+version=0.77.0
+revision=1
+build_style=meson
+hostmakedepends="pkg-config ccache"
```
Comment (ericonr) [inline srcpkgs/dosbox-staging/template:6]: `ccache` is always available in the chroot, don't install manually.

File: srcpkgs/libmt32emu/template [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+homepage="http://munt.sourceforge.net"
+distfiles="https://github.com/munt/munt/archive/libmt32emu_${_version_string}.tar.gz"
+checksum=684988ab4d2b103554d9a38ad556cebc3ebc28d1d8fbb49fb4808ec9c80a2bf4
+
+build_options="libsoxr libsamplerate"
+desc_option_libsoxr="Use libsoxr for sample rate conversion"
+desc_option_libsamplerate="Use libsamplerate for sample rate conversion"
+build_options_default="libsoxr"
+vopt_conflict libsoxr libsamplerate
+if [ "$build_option_libsoxr" -o "$build_option_libsamplerate" ]; then configure_args+=" -Dlibmt32emu_WITH_INTERNAL_RESAMPLER=off"; fi
```
Comment (ericonr) [inline srcpkgs/libmt32emu/template:22]: Everything after `then` should be a new line, and so should `fi`.

File: srcpkgs/libmt32emu/template [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+libmt32emu-devel_package() {
+	short_desc+=" - development files"
+	depends="${sourcepkg}>=${version}_${revision} $(vopt_if libsoxr libsoxr-devel) $(vopt_if libsamplerate libsamplerate-devel)"
```
Comment (ericonr) [inline srcpkgs/libmt32emu/template:26]:
```suggestion
	depends="${sourcepkg}>=${version}_${revision} ${makedepends}"
```

File: srcpkgs/dosbox-staging/template (hunk 1 above)
Comment (joshuakraemer) [reply srcpkgs/dosbox-staging/template]: Done and check for `XBPS_TARGET_NO_ATOMIC8` added.

File: srcpkgs/dosbox-staging/template (hunk 2 above)
Comment (joshuakraemer) [reply srcpkgs/dosbox-staging/template]: Done.

File: srcpkgs/libmt32emu/template (hunk 3 above)
Comment (joshuakraemer) [reply srcpkgs/libmt32emu/template]: Done.

File: srcpkgs/libmt32emu/template (hunk 4 above)
Comment (joshuakraemer) [reply srcpkgs/libmt32emu/template]: Done.

**Comment by ericonr** [comment]
```
Files of libmt32emu:
/usr/lib/libmt32emu.so.2.5.1
/usr/share/doc/munt/libmt32emu/AUTHORS.txt
/usr/share/doc/munt/libmt32emu/COPYING.LESSER.txt
/usr/share/doc/munt/libmt32emu/COPYING.txt
/usr/share/doc/munt/libmt32emu/NEWS.txt
/usr/share/doc/munt/libmt32emu/README.md
/usr/share/doc/munt/libmt32emu/TODO.txt
/usr/lib/libmt32emu.so.2 -> /usr/lib/libmt32emu.so.2.5.1
```

I think the files under `/usr/share/doc` can be removed, can't they?

File: srcpkgs/libmt32emu/template [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+desc_option_libsamplerate="Use libsamplerate for sample rate conversion"
+build_options_default="libsoxr"
+vopt_conflict libsoxr libsamplerate
+
+if [ "$build_option_libsoxr" -o "$build_option_libsamplerate" ]; then
+	configure_args+=" -Dlibmt32emu_WITH_INTERNAL_RESAMPLER=off"
+fi
+
+post_install() {
+	rm -rf ${DESTDIR}/usr/share/doc
```
Comment (ericonr) [inline srcpkgs/libmt32emu/template:30]: Use `-r` so we get an error if they stop installing the files.

File: srcpkgs/libmt32emu/template [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+version=2.5.1
+revision=1
+_version_string=${version//./_}
+wrksrc=munt-libmt32emu_${_version_string}
+build_wrksrc=mt32emu
+build_style=cmake
+makedepends="$(vopt_if libsoxr libsoxr-devel)
+ $(vopt_if libsamplerate libsamplerate-devel)"
+short_desc="Library to emulate Roland MT-32, CM-32L and LAPC-I synthesizer
+ modules"
```
Comment (ericonr) [inline srcpkgs/libmt32emu/template:12]:
```suggestion
short_desc="Emulate Roland MT-32, CM-32L and LAPC-I synthesizer modules"
```

If `short_desc` is big enough to need to be split, it should be cut down instead :)

File: srcpkgs/libmt32emu/template (hunk 5 above)
Comment (joshuakraemer) [reply srcpkgs/libmt32emu/template]: Done

File: srcpkgs/libmt32emu/template (hunk 6 above)
Comment (joshuakraemer) [reply srcpkgs/libmt32emu/template]: Done

**Comment by ericonr** [comment]
Re-ordered commits for logical order (first dependency, then dosbox).

**Pull request merged**